exporter/opentelemetry-exporter-otlp-proto-common/src/opentelemetry/exporter/otlp/proto/common/_internal/_log_encoder/__init__.py

```diff
@@ -39,11 +39,21 @@ def encode_logs(batch: Sequence[LogData]) -> ExportLogsServiceRequest:
 
 
 def _encode_log(log_data: LogData) -> PB2LogRecord:
+    span_id = (
+        None
+        if log_data.log_record.span_id == 0
+        else _encode_span_id(log_data.log_record.span_id)
+    )
+    trace_id = (
+        None
+        if log_data.log_record.trace_id == 0
+        else _encode_trace_id(log_data.log_record.trace_id)
+    )
     return PB2LogRecord(
         time_unix_nano=log_data.log_record.timestamp,
         observed_time_unix_nano=log_data.log_record.observed_timestamp,
-        span_id=_encode_span_id(log_data.log_record.span_id),
-        trace_id=_encode_trace_id(log_data.log_record.trace_id),
+        span_id=span_id,
```

Comment:
Just curious but does `trace_id` and `span_id` get included in the serialization if they are set to `None`? What is the behavior that shown in the serialized protobuf?

Reply:
They don't get included when set to None. This is what it looks like for the 'example.py' script from the issue with slight tweaks:
```
{
  "resourceLogs": [
    {
      "resource": {
        "attributes": [
          {
            "key": "telemetry.sdk.language",
            "value": {
              "stringValue": "python"
            }
          },
          {
            "key": "telemetry.sdk.name",
            "value": {
              "stringValue": "opentelemetry"
            }
          },
          {
            "key": "telemetry.sdk.version",
            "value": {
              "stringValue": "1.26.0.dev0"
            }
          },
          {
            "key": "service.name",
            "value": {
              "stringValue": "shoppingcart"
            }
          },
          {
            "key": "service.instance.id",
            "value": {
              "stringValue": "instance-12"
            }
          }
        ]
      },
      "scopeLogs": [
        {
          "scope": {
            "name": "opentelemetry.sdk._logs._internal"
          },
          "logRecords": [
            {
              "timeUnixNano": "1719860398396537088",
              "severityNumber": "SEVERITY_NUMBER_WARN",
              "severityText": "WARN",
              "body": {
                "stringValue": "Jail zesty vixen who grabbed pay from quack."
              },
              "attributes": [
                {
                  "key": "code.filepath",
                  "value": {
                    "stringValue": "<path to example.py>/example.py"
                  }
                },
                {
                  "key": "code.function",
                  "value": {
                    "stringValue": "<module>"
                  }
                },
                {
                  "key": "code.lineno",
                  "value": {
                    "intValue": "48"
                  }
                }
              ],
              "observedTimeUnixNano": "1719860398396555000"
            },
            {
              "timeUnixNano": "1719860398396589056",
              "severityNumber": "SEVERITY_NUMBER_ERROR",
              "severityText": "ERROR",
              "body": {
                "stringValue": "The five boxing wizards jump quickly."
              },
              "attributes": [
                {
                  "key": "code.filepath",
                  "value": {
                    "stringValue": "<path to example.py>/example.py"
                  }
                },
                {
                  "key": "code.function",
                  "value": {
                    "stringValue": "<module>"
                  }
                },
                {
                  "key": "code.lineno",
                  "value": {
                    "intValue": "49"
                  }
                }
              ],
              "observedTimeUnixNano": "1719860398396595000"
            },
            {
              "timeUnixNano": "1719860398396794880",
              "severityNumber": "SEVERITY_NUMBER_ERROR",
              "severityText": "ERROR",
              "body": {
                "stringValue": "Hyderabad, we have a major problem."
              },
              "attributes": [
                {
                  "key": "code.filepath",
                  "value": {
                    "stringValue": "<path to example.py>/example.py"
                  }
                },
                {
                  "key": "code.function",
                  "value": {
                    "stringValue": "<module>"
                  }
                },
                {
                  "key": "code.lineno",
                  "value": {
                    "intValue": "56"
                  }
                }
              ],
              "flags": 1,
              "traceId": "1zepdflw89AcMmOKaPX6dQ==",
              "spanId": "SY7ufwvDWqE=",
              "observedTimeUnixNano": "1719860398396802000"
            }
          ]
        }
      ]
    }
  ]
}
```